### PR TITLE
[REG-975] feat: Added Blockchain Family Keys to ProxyReader

### PR DESCRIPTION
**Pull request by nickshatilo**
### 1. Contracts versioning
- [x] Make sure that the `patch` version of the contracts is increased if changes have been made to the `UNSRegistry`, `MintingManager` or `ProxyReader` contracts.
- [ ] Make sure that the `minor` version of the contracts is increased if breaking changes have been made to the `UNSRegistry`, `MintingManager` or `ProxyReader` contracts. It includes changes of interfaces.
### 2. Contracts licensing
- [x] Make sure that no **SPDX-License-Identifier** defined in contracts.
- [ ] Make sure that the **header** is added to the new contract files. 
  ```
  // @author Unstoppable Domains, Inc.
  // @date {Month} {Day}(ordinal), {Year}
  ```
### 3. Coverage
- [ ] Make sure that the coverage of contracts has not decreased and strive **100%**
### 4. Configs versioning
- [ ] Make sure that the version of `uns-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `resolver-keys.json` is increased if changes have been made to the config.
### 5. Package versioning
- [ ] Make sure that the `patch` version of package is increased if valuable changes have been made to the package. It includes contracts update, configs update, etc.
- [x] Make sure that the `major.minor` version of package is synced with version of `UNSRegistry` contract.
- [x] Make sure that the `CHANGELOG` is updated with short description for the new version. 
### 6. Code review
- [ ] `resolver-keys.json` code review is required from **DevTools** team
- [x] For all other changes code review is required from **Registry** team

### CODE REVIEW

File: contracts/ProxyReader.sol [hunk 1]
```diff
@@ -247,10 +299,50 @@ contract ProxyReader is ERC165Upgradeable, MulticallUpgradeable, IRegistryReader
...
+    }
+
+    function addBlockchainNetworksV2(string[] calldata networks, string calldata family) external onlyOwner {
+        for (uint256 i = 0; i < networks.length; i++) {
+            _families[networks[i]] = family;
+        }
+    }
+
+    function addLegacyRecords(string[] calldata keys, string[][] calldata legacyKeys) external onlyOwner {
+        _ensureSameLength(keys.length, legacyKeys.length);
```
Comment (aquiladev) [inline contracts/ProxyReader.sol:317]: I would not use function, it might increase gas usage

File: contracts/ProxyReader.sol [hunk 2]
```diff
@@ -247,10 +299,50 @@ contract ProxyReader is ERC165Upgradeable, MulticallUpgradeable, IRegistryReader
...
+
+        for (uint256 i = 0; i < keys.length; i++) {
+            delete _legacyKeys[keys[i]];
```
Comment (aquiladev) [inline contracts/ProxyReader.sol:320]: I'm thinking that `_legacyKeys` might have more items then `keys.length`
not sure how big the problem

File: contracts/utils/Ownable.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        emit OwnershipTransferred(oldOwner, newOwner);
+    }
+
+    function _msgSender() internal view virtual returns (address) {
+        return msg.sender;
+    }
+
+    function _msgData() internal view virtual returns (bytes calldata) {
+        return msg.data;
+    }
```
[the remark this answers is not in the document]
Comment (nickshatilo) [reply contracts/utils/Ownable.sol:98]: I rechecked it and unfortunately, it does. `ContextUpgradable` has a gap.

File: contracts/ProxyReader.sol (hunk 2 above)
Comment (nickshatilo) [reply contracts/ProxyReader.sol:320]: Closing per our discussion

File: test/ProxyReader.test.ts [hunk 4]
```diff
@@ -985,6 +985,260 @@ describe('ProxyReader', () => {
...
+      it('should return empty string if token does not exist', async () => {
+        expect(
+          await proxyReader.getAddressKey('ETH', 'USDC', 0x42),
+        ).to.equal('');
+      });
+
+      it('should return empty string if family is not defined', async () => {
+        expect(
+          await proxyReader.getAddressKey('UDTOKEN', 'UDTOKEN', walletTokenId),
+        ).to.equal('');
```
Comment (DeRain) [inline test/ProxyReader.test.ts:1237]: Nice to have some tests with some skipped keys/levels

**Review by DeRain — APPROVED** [review]
Looks great to me!

File: contracts/ProxyReader.sol [hunk 5]
```diff
@@ -335,6 +434,28 @@ contract ProxyReader is ERC165Upgradeable, MulticallUpgradeable, IRegistryReader
...
+        for (uint256 i = 0; i < records.length; i++) {
+            if (bytes(records[i]).length > 0) {
+                key = keys[i];
+                addr = records[i];
+                break;
+            }
+        }
+    }
+
+    function _setNetworkfamily(string calldata network, string calldata family) private {
```
Comment (aquiladev) [inline contracts/ProxyReader.sol:454]:
```suggestion
    function _setNetworkFamily(string calldata network, string calldata family) private {
```

File: contracts/ProxyReader.sol [hunk 6]
```diff
@@ -247,10 +302,54 @@ contract ProxyReader is ERC165Upgradeable, MulticallUpgradeable, IRegistryReader
         return address(0);
     }
 
+    function addBlockchainNetworksV1(string[] calldata networks, string[] calldata families) external onlyOwner {
```
Comment (aquiladev) [inline contracts/ProxyReader.sol:305]: do we really need `V1`,`V2` postfixes?

**Review by aquiladev — APPROVED** [review]

File: test/ProxyReader.test.ts (hunk 4 above)
Comment (nickshatilo) [reply test/ProxyReader.test.ts:1361]: Added :)

**Pull request merged**